### PR TITLE
append to an array by emplacement

### PR DESCRIPTION
**Pull request by wilzbach**
Reboot of @cyisfor's #4116 (with tabs -> spaces and smaller style fixes)

### CODE REVIEW

**Comment by andralex** [comment]
This adds an undocumented public function. Also there seems to be no bounds checking - what gives?

File: std/array.d [hunk 1]
```diff
@@ -2844,6 +2844,19 @@ if (isDynamicArray!A)
         }
     }
 
+    // emplace an item onto an array.
+    void emplacePut(Args...)(Args args) {
+        import std.conv : emplaceRef;
+
```
Comment (andralex) [inline std/array.d]: remove fluff

File: std/array.d [hunk 2]
```diff
@@ -2844,6 +2844,19 @@ if (isDynamicArray!A)
...
+        ensureAddable(1);
+        immutable len = _data.arr.length;
+
```
Comment (andralex) [inline std/array.d]: remove fluff

File: std/array.d [hunk 3]
```diff
@@ -2844,6 +2844,19 @@ if (isDynamicArray!A)
...
+        auto bigData = (() @trusted => _data.arr.ptr[0..len + 1])();
```
Comment (andralex) [inline std/array.d]: no bounds check
space around `..`

**Comment by andralex** [comment]
(We really should add a style rule that vertical space has no business in a function under 10 lines.)

**Comment by wilzbach** [comment]
Closing this as I have just revived this PR from the dead and lost the time to maintain it.

**Pull request closed without merging**